tools/tplg_parser/CMakeLists.txt

```diff
@@ -0,0 +1,32 @@
+# SPDX-License-Identifier: BSD-3-Clause
+
+cmake_minimum_required(VERSION 3.10)
+
+project(SOF_TPLG_PARSER C)
+
+set(sof_source_directory "${PROJECT_SOURCE_DIR}/../..")
+
+add_library(sof_parser_inc INTERFACE)
+target_include_directories(sof_parser_inc INTERFACE ${CMAKE_CURRENT_SOURCE_DIR}/include)
+
+add_library(sof_tplg_parser SHARED tplg_parser.c)
+target_include_directories(sof_tplg_parser PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/include)
+target_include_directories(sof_tplg_parser PRIVATE ${sof_source_directory}/src/include)
+
+get_target_property(incdirs sof_parser_inc INTERFACE_INCLUDE_DIRECTORIES)
+
+# we append slash at the end to make CMake copy contents of directories
+# instead of directories
+set(incdirs_for_install "")
+
+foreach(d ${incdirs})
+	list(APPEND incdirs_for_install "${d}/")
+endforeach()
+
+install(DIRECTORY ${incdirs_for_install}
```

Comment:
once you remove foreach and sof_parser_inc target, you can just use `${CMAKE_CURRENT_SOURCE_DIR}/include` here